### PR TITLE
Max compilation update

### PR DESCRIPTION
**Pull request by BraveMole**
Warn the developpers when not on 514.1557 or higher when compiler the codebase

Use 514.1557 when compiling on CL

### CODE REVIEW

**Comment by SplinterGP** [comment]
test 1557

**Comment by ZephyrTFA** [comment]
> test 1557

1557 works for us upstream

File: code/_compile_options.dm [hunk 1]
```diff
@@ -38,7 +38,7 @@
 
 //Update this whenever the byond version is stable so people stop updating to hilariously broken versions
 #define MAX_COMPILER_VERSION 514
-#define MAX_COMPILER_BUILD 1554
+#define MAX_COMPILER_BUILD 1556
```
Comment (SplinterGP) [inline code/_compile_options.dm:41]:
```suggestion
#define MAX_COMPILER_BUILD 1557
```

**Review by SplinterGP — CHANGES_REQUESTED** [review]

**Comment by spookydonut** [comment]
https://github.com/tgstation/TerraGov-Marine-Corps/blob/master/dependencies.sh

**Comment by TiviPlus** [comment]
You need to make it pass ci

File: code/_compile_options.dm [hunk 2]
```diff
@@ -26,19 +26,19 @@
 #endif
 
 //Update this whenever you need to take advantage of more recent byond features
-#define MIN_COMPILER_VERSION 513
-#define MIN_COMPILER_BUILD 1523
+#define MIN_COMPILER_VERSION 514
+#define MIN_COMPILER_BUILD 1557
```
Comment (SplinterGP) [inline code/_compile_options.dm:30]: why did you increase the minimum to 514 ?

**Review by SplinterGP — CHANGES_REQUESTED** [review]

File: code/modules/unit_tests/xeno_logical_scaling.dm [hunk 3]
```diff
@@ -29,35 +31,17 @@
 		var/list/lesser_test_vars = list(
 			"speed" = 99,
 		)
+		for(var/upgradepath in mob_data)
+			var/fullxenopath = mob_data[upgradepath]
+			var/mob/living/carbon/xenomorph/xeno_mob = new fullxenopath()
```
Comment (TiviPlus) [inline code/modules/unit_tests/xeno_logical_scaling.dm:36]: use allocate

File: code/_compile_options.dm [hunk 4]
```diff
@@ -26,19 +26,19 @@
 #endif
 
 //Update this whenever you need to take advantage of more recent byond features
-#define MIN_COMPILER_VERSION 513
-#define MIN_COMPILER_BUILD 1523
+#define MIN_COMPILER_VERSION 514
```
Comment (TiviPlus) [inline code/_compile_options.dm:29]: Dont update minimum versions unless it is explicitly to make use of new features

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Comment by spookydonut** [comment]
Yeah no

**Pull request closed without merging**